### PR TITLE
Bug fix (Unity): GenerateErrorReport can provide incomplete and misleading information…

### PR DESCRIPTION
**Pull request by JTonoJimenez**
… (Unity)

### CODE REVIEW

**Comment by JTonoJimenez** [comment]
Unity Solution for the same problem as https://github.com/PlayFab/SDKGenerator/pull/550

**Comment by Vicodex** [comment]
retest

File: targets/unity-v2/source/Shared/Internal/PlayFabErrors.cs.ejs [hunk 1]
```diff
@@ -41,12 +41,25 @@ namespace PlayFab
...
+        /// <returns>A description of the error that we just incur.</returns>
         public string GenerateErrorReport()
         {
             if (_tempSb == null)
                 _tempSb = new StringBuilder();
             _tempSb.Length = 0;
-            _tempSb.Append(ApiEndpoint).Append(": ").Append(ErrorMessage);
+            if (String.IsNullOrEmpty(ErrorMessage))
+            {
+                _tempSb.Append("Http Code: ").Append(HttpCode.ToString()).Append("\nHttp Status: ").Append(HttpStatus).Append("\nError: ").Append(Error.ToString()).Append("\n");
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabErrors.cs.ejs]: This line should also likely include ApiEndpoint.
ApiEndpoint is the destination URL, and is also present if the result is (404, etc)

**Comment by pgilmorepf** [comment]
retest

**Comment by Vicodex** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
FINALLY!

**Pull request merged**